--- ./rpcs3/Emu/RSX/Common/texture_cache.h
+++ ./rpcs3/Emu/RSX/Common/texture_cache.h
@@ -2790,6 +2790,9 @@
 
 				lock.upgrade();
 
+			if (g_cfg.video.reprotect_hack)
+				cached_dest->reprotect(utils::protection::ro, { mem_offset, dst_payload_length });
+			else
 				cached_dest->reprotect(utils::protection::no, { mem_offset, dst_payload_length });
 				cached_dest->touch(m_cache_update_tag);
 				update_cache_tag();

