--- ./rpcs3/rpcs3qt/tooltips.h
+++ ./rpcs3/rpcs3qt/tooltips.h
@@ -91,6 +91,8 @@
 		const QString force_cpu_blit_emulation     = tr("Forces emulation of all blit and image manipulation operations on the CPU.\nRequires 'Write Color Buffers' option to also be enabled in most cases to avoid missing graphics.\nSignificantly degrades performance but is more accurate in some cases.\nThis setting overrides the 'GPU texture scaling' option.");
 		const QString disable_vulkan_mem_allocator = tr("Disables the custom Vulkan memory allocator and reverts to direct calls to VkAllocateMemory/VkFreeMemory.");
 		const QString disable_fifo_reordering      = tr("Disables RSX FIFO optimizations completely. Draws are processed as they are received by the DMA puller.");
+		const QString reprotect_hack               = tr("Never use this.\nDisable Force CPU Blit when using this option.\nWorkaround for flickering shadows and missing water in TLoU, Uncharted 3.");
+		const QString vk_event_hack                = tr("Improves performance where Write Color Buffer or Write Depth Buffer is used\nIf unsure never use this.");
 		const QString gpu_texture_scaling          = tr("Force all texture transfer, scaling and conversion operations on the GPU.\nMay cause texture corruption in some cases.");
 		const QString strict_texture_flushing      = tr("Forces texture flushing even in situations where it is not necessary/correct. Known to cause visual artifacts, but useful for debugging certain texture cache issues.");
 		const QString disable_native_fp16          = tr("Disables hardware half-float support which is known to cause problems in some rare cases on some GPUs.");

